src/addon/info.cpp

```diff
@@ -132,6 +151,68 @@ std::string addon_info::display_title() const
 	}
 }
 
+addon_info_translation addon_info_translation::invalid = {"en_US", false, "invalid_addon!", ""};
+
+addon_info_translation addon_info::translated_info() const
+{
+	std::string locale = get_language().localename;
+
+	if(locale != "en_US") {
+		for(const addon_info_translation& info : this->info_translations) {
+			if(info.language == locale || info.language == locale.substr(0, 2)) {
+				return info;
+			}
```

Comment:
Is there a reason why this seems to accept `[translation]`s for language variants that may or may not be actually compatible (such as the different Chinese variants)? Also noting that the order in which they are found is not guaranteed AFAICT — if the .pbl file is preseeded with `[translation]` blocks then I figure the author can make it so any other variant is selected here instead of the language that the player is actually using, even if both are supported.

Reply:
I think it can be done with two loops in that case. The first will look for the full locale, the second for the short version.

Reply:
The fact that you need to do any passes at all shows that the the design may not be terribly optimal (not that it matters too much in this code path since it's client-side and not a particularly hot one). Personally I would've made `addon_info::info_translations` a `std::map` indexed by locale ids to save myself the trouble of having to go over the full list every time I'm looking for a particular entry.